### PR TITLE
 modify log of sync && fix sync logic

### PR DESCRIPTION
**Pull request by cyjseagull**
modify log of sync && fix sync logic

### CODE REVIEW

File: libsync/SyncMaster.cpp [hunk 1]
```diff
@@ -36,22 +36,22 @@ static unsigned const c_maxSendTransactions = 128;
...
-    SYNC_LOG(TRACE) << "            IsSyncing:    " << isSyncing() << endl;
-    SYNC_LOG(TRACE) << "            Block number: " << m_blockChain->number() << endl;
+    SYNC_LOG(TRACE) << "[Sync Info] --------------------------------------------";
+    SYNC_LOG(TRACE) << "            IsSyncing:    " << isSyncing();
+    SYNC_LOG(TRACE) << "            Block number: " << m_blockChain->number();
     SYNC_LOG(TRACE) << "            Block hash:   "
-                    << m_blockChain->numberHash(m_blockChain->number()) << endl;
-    SYNC_LOG(TRACE) << "            Genesis hash: " << m_syncStatus->genesisHash << endl;
+                    << m_blockChain->numberHash(m_blockChain->number());
+    SYNC_LOG(TRACE) << "            Genesis hash: " << m_syncStatus->genesisHash.abridged();
```
Comment (bxq2011hust) [inline libsync/SyncMaster.cpp]: could we change the log to one line to avoid multi threads unordered log?

File: libsync/SyncMaster.cpp [hunk 2]
```diff
@@ -499,20 +506,29 @@ void SyncMaster::maintainPeersConnection()
...
+    if (hasMyself)
     {
-        if (memberSet.find(id) == memberSet.end())
-            m_syncStatus->deletePeer(id);
+        NodeIDs nodeIds = m_syncStatus->peers();
+        for (NodeID const& id : nodeIds)
+        {
+            if (memberSet.find(id) == memberSet.end())
+            {
+                m_syncStatus->deletePeer(id);
```
Comment (wheatli) [inline libsync/SyncMaster.cpp]: break here ? to avoid traversing the list

File: libsync/SyncMaster.cpp (hunk 1 above)
Comment (cyjseagull) [reply libsync/SyncMaster.cpp]: done

File: libsync/SyncMaster.cpp (hunk 2 above)
Comment (cyjseagull) [reply libsync/SyncMaster.cpp]: Add `break` here will change the logic.

**Pull request merged**